src/core/src/service_interfaces/StatusHandler.py

```diff
@@ -795,14 +842,348 @@ def __try_add_error(error_list, detail):
+                                __recompose_substatus_msg_errors()
+                                __create_assessment_tombstone_list()
+                                    __create_assessment_tombstone()
+                                __recreate_assessment_summary_json()
+                       
+                        __recompose_truncated_status_file()                                    
+                            __get_current_complete_status_errors()
+                            __recompose_truncated_substatus_msg()
+                                __recompose_substatus_msg_errors()
+                                __create_installation_tombstone
+                                __recreate_installation_summary_json()
+                                
+                        status_file_size_in_bytes, status_file_agent_size_diff = __get_new_size_in_bytes_after_truncation(truncated_status_file)
+                        size_of_max_packages_allowed_in_status -= status_file_agent_size_diff  
+        """
+        self.composite_logger.log_debug("Begin package list truncation")
+        truncated_status_file = json.loads(complete_status_file_payload)  # reload payload into python object
+        low_pri_index = None
+        assessment_substatus_index = self.__get_substatus_index(Constants.PATCH_ASSESSMENT_SUMMARY, truncated_status_file['status']['substatus'])
+        installation_substatus_index = self.__get_substatus_index(Constants.PATCH_INSTALLATION_SUMMARY, truncated_status_file['status']['substatus'])
+
+        if assessment_substatus_index is not None:        # if assessment data exists
+            self.__assessment_substatus_msg_copy = self.__get_substatus_msg_at_index(truncated_status_file, assessment_substatus_index)
+            self.__assessment_packages_copy = self.__assessment_substatus_msg_copy['patches']
+
+        if installation_substatus_index is not None:      # if installation data exists
+            self.__installation_substatus_msg_copy = self.__get_substatus_msg_at_index(truncated_status_file, installation_substatus_index)
+            self.__installation_packages_copy = self.__installation_substatus_msg_copy['patches']
+            low_pri_index = self.__get_installation_low_pri_index(self.__installation_packages_copy)
+
+        status_file_without_package_list_size = self.size_of_constant_status_data(copy.deepcopy(truncated_status_file), assessment_substatus_index, installation_substatus_index)  # deepcopy fully copy the object avoid reference modification
+        size_of_max_packages_allowed_in_status = self.__internal_file_capacity - status_file_without_package_list_size
+
+        while status_file_size_in_bytes > self.__internal_file_capacity:
+            # Start truncation process
+            packages_retained_in_assessment, assessment_packages_removed_after_truncation, packages_retained_in_installation, installation_packages_removed_after_truncation = \
```

Comment:
keep naming structure similar. For eg: if using 'packages_retained_in_assessment' the other variable should be 'packages_removed_from_assessment'.